src/core/qgsmaprenderertask.h

```diff
@@ -60,11 +61,18 @@ class CORE_EXPORT QgsMapRendererTask : public QgsTask
      * If the output \a fileFormat is set to PDF, the \a geoPdf argument controls whether a GeoPDF file is created.
      * See QgsAbstractGeoPdfExporter::geoPDFCreationAvailable() for conditions on GeoPDF creation availability.
      */
+#ifndef SIP_RUN
     QgsMapRendererTask( const QgsMapSettings &ms,
                         const QString &fileName,
                         const QString &fileFormat = QString( "PNG" ),
                         bool forceRaster = false,
-                        bool geoPdf = false );
+                        bool geoPdf = false, const QgsAbstractGeoPdfExporter::ExportDetails &geoPdfExportDetails = QgsAbstractGeoPdfExporter::ExportDetails() );
```

Comment:
This essentially means using the map renderer task via python won't be able to output GeoPDF-formatted PDFs, is there anything technically that prevents this?

Reply:
Yes - I don't want to expose QgsAbstractGeoPdfExporter to Python yet. I'd rather give it a few cycles for the API to stablise.